### PR TITLE
Fix to convert multiple CryoSat2 sea-ice freeboard data files into one concatenated daily file

### PR DESCRIPTION
**Pull request by mjkagnes123**
Changes to be able to convert multiple CryoSat2 sea-ice freeboard data files into one concatenated daily file

Issue related with this PR
  #457

### CODE REVIEW

**Comment by HamidehGMAO** [comment]
@mjkagnes123 could you please merge develop to your branch, the aws tests are failing

**Comment by HamidehGMAO** [comment]
@mjkagnes123 the CryoSat2 test is failing

**Comment by srherbener** [comment]
@mjkagnes123 here are the error messages from the CI tests:
```
test 5
888 | Start  5: iodaconv_marine_coding_norms
889 |  
890 | 5: Test command: /build_container/bin/iodaconv_lint.sh "/jcsda/ioda-bundle/iodaconv/src/marine" "/jcsda/ioda-bundle/iodaconv"
891 | 5: Environment variables:
892 | 5:  OMP_NUM_THREADS=1
893 | 5: Test timeout computed to be: 10000000
894 | 5: + SRC_DIR=/jcsda/ioda-bundle/iodaconv/src/marine
895 | 5: + PYCODESTYLE_CFG_PATH=/jcsda/ioda-bundle/iodaconv
896 | 5: + cd /jcsda/ioda-bundle/iodaconv/src/marine
897 | 5: + pycodestyle -v --config=/jcsda/ioda-bundle/iodaconv/.pycodestyle --filename=*.py,*.py.in .
898 | 5: cli configuration: /jcsda/ioda-bundle/iodaconv/.pycodestyle
899 | 5: directory .
900 | 5: checking ./argoClim2ioda.py
901 | 5: checking ./cryosat_ice2ioda.py
902 | 5: ./cryosat_ice2ioda.py:38:34: W291 trailing whitespace
903 | 5: ./cryosat_ice2ioda.py:46:47: W291 trailing whitespace
904 | 5: ./cryosat_ice2ioda.py:47:47: W291 trailing whitespace
905 | 5: ./cryosat_ice2ioda.py:48:45: W291 trailing whitespace
906 | 5: ./cryosat_ice2ioda.py:51:12: E111 indentation is not a multiple of four
907 | 5: ./cryosat_ice2ioda.py:52:12: E111 indentation is not a multiple of four
908 | 5: ./cryosat_ice2ioda.py:53:12: E111 indentation is not a multiple of four
909 | 5: ./cryosat_ice2ioda.py:54:12: E111 indentation is not a multiple of four
910 | 5: ./cryosat_ice2ioda.py:55:12: E111 indentation is not a multiple of four
911 | 5: ./cryosat_ice2ioda.py:56:12: E111 indentation is not a multiple of four
912 | 5: ./cryosat_ice2ioda.py:57:12: E111 indentation is not a multiple of four
913 | 5: ./cryosat_ice2ioda.py:58:12: E111 indentation is not a multiple of four
914 | 5: ./cryosat_ice2ioda.py:59:12: E111 indentation is not a multiple of four
915 | 5: ./cryosat_ice2ioda.py:60:12: E114 indentation is not a multiple of four (comment)
916 | 5: ./cryosat_ice2ioda.py:61:12: E111 indentation is not a multiple of four
917 | 5: ./cryosat_ice2ioda.py:62:12: E122 continuation line missing indentation or outdented
918 | 5: ./cryosat_ice2ioda.py:63:12: E111 indentation is not a multiple of four
919 | 5: ./cryosat_ice2ioda.py:64:12: E111 indentation is not a multiple of four
920 | 5: ./cryosat_ice2ioda.py:66:12: E114 indentation is not a multiple of four (comment)
921 | 5: ./cryosat_ice2ioda.py:67:12: E111 indentation is not a multiple of four
922 | 5: ./cryosat_ice2ioda.py:68:16: E111 indentation is not a multiple of four
923 | 5: ./cryosat_ice2ioda.py:69:16: E111 indentation is not a multiple of four
924 | 5: ./cryosat_ice2ioda.py:70:16: E111 indentation is not a multiple of four
925 | 5: ./cryosat_ice2ioda.py:71:16: E111 indentation is not a multiple of four
926 | 5: ./cryosat_ice2ioda.py:72:16: E111 indentation is not a multiple of four
927 | 5: ./cryosat_ice2ioda.py:73:16: E111 indentation is not a multiple of four
928 | 5: ./cryosat_ice2ioda.py:74:16: E111 indentation is not a multiple of four
929 | 5: ./cryosat_ice2ioda.py:76:12: E111 indentation is not a multiple of four
930 | 5: ./cryosat_ice2ioda.py:77:16: E111 indentation is not a multiple of four
931 | 5: ./cryosat_ice2ioda.py:77:27: E222 multiple spaces after operator
932 | 5: ./cryosat_ice2ioda.py:77:35: E203 whitespace before ':'
933 | 5: ./cryosat_ice2ioda.py:77:40: E262 inline comment should start with '# '
934 | 5: ./cryosat_ice2ioda.py:78:18: E111 indentation is not a multiple of four
935 | 5: ./cryosat_ice2ioda.py:79:18: E111 indentation is not a multiple of four
936 | 5: ./cryosat_ice2ioda.py:80:18: E111 indentation is not a multiple of four
937 | 5: ./cryosat_ice2ioda.py:81:18: E111 indentation is not a multiple of four
938 | 5: ./cryosat_ice2ioda.py:82:18: E111 indentation is not a multiple of four
939 | 5: ./cryosat_ice2ioda.py:84:12: E111 indentation is not a multiple of four
940 | 5: checking ./emc_ice2ioda.py
941 | 5: checking ./gds2_sst2ioda.py
942 | 5: checking ./glider2ioda.py
943 | 5: checking ./gmao_obs2ioda.py
944 | 5: checking ./godae_profile2ioda.py
945 | 5: checking ./godae_ship2ioda.py
946 | 5: checking ./godae_trak2ioda.py
947 | 5: checking ./hgodas_adt2ioda.py
948 | 5: checking ./hgodas_insitu2ioda.py
949 | 5: checking ./hgodas_sst2ioda.py
950 | 5: checking ./rads_adt2ioda.py
951 | 5: checking ./smap_sss2ioda.py
952 | 5: checking ./viirs_modis_oc2ioda.py
953 | 5: 28      E111 indentation is not a multiple of four
954 | 5: 2       E114 indentation is not a multiple of four (comment)
955 | 5: 1       E122 continuation line missing indentation or outdented
956 | 5: 1       E203 whitespace before ':'
957 | 5: 1       E222 multiple spaces after operator
958 | 5: 1       E262 inline comment should start with '# '
959 | 5: 4       W291 trailing whitespace
960 | 1/2 Test  #5: iodaconv_marine_coding_norms .....***Failed    0.32 sec
961 | test 24
962 | Start 24: test_iodaconv_cryosat2
963 |  
964 | 24: Test command: /build_container/bin/iodaconv_comp.sh "netcdf" "/build_container/bin/cryosat_ice2ioda.py
965 | -i testinput/cryosat2_L2_test.nc
966 | -o testrun/cryosat2_L2.nc
967 | -d 2019092112" "cryosat2_L2.nc" "0.0"
968 | 24: Environment variables:
969 | 24:  OMP_NUM_THREADS=1
970 | 24: Test timeout computed to be: 10000000
971 | 24: 2019-09-21 12:00:00
972 | 24: testinput/cryosat2_L2_test.nc
973 | 24: DIFFER : VALUES OF GLOBAL ATTRIBUTE : nlocs : 1 <> 87
974 | 24: DIFFER : LENGTHS : DIMENSION : nlocs : 1 <> 87
975 | 24: DIFFER : VARIABLE : sea_ice_freeboard@ObsValue : SIZE : 1 <> 87
976 | 24: DIFFER : VARIABLE : sea_ice_freeboard@ObsError : SIZE : 1 <> 87
977 | 24: DIFFER : VARIABLE : sea_ice_freeboard@PreQC : SIZE : 1 <> 87
978 | 24: DIFFER : VARIABLE : time@MetaData : SIZE : 1 <> 87
979 | 24: DIFFER : VARIABLE : latitude@MetaData : SIZE : 1 <> 87
980 | 24: DIFFER : VARIABLE : longitude@MetaData : SIZE : 1 <> 87
981 | 24: DIFFER : VARIABLE : datetime@MetaData : SIZE : 20 <> 1740
982 | 24: DIFFER : VARIABLE : record_number@MetaData : SIZE : 1 <> 87
983 | 2/2 Test #24: test_iodaconv_cryosat2 ...........***Failed    0.31 sec
984 |  
985 | 0% tests passed, 2 tests failed out of 2
986 |  
987 | Label Time Summary:
988 | iodaconv    =   0.63 sec*proc (2 tests)
989 | script      =   0.63 sec*proc (2 tests)
990 |  
991 | Total Test time (real) =   0.63 sec
992 |  
993 | The following tests FAILED:
994 | 5 - iodaconv_marine_coding_norms (Failed)
995 | 24 - test_iodaconv_cryosat2 (Failed)
996 | Errors while running CTest
997
```

File: src/marine/cryosat_ice2ioda.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import os
 from pathlib import Path
 
-IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+IODA_CONV_PATH = Path(__file__).parent/"../lib/pyiodaconv"
```
Comment (markjolah) [inline src/marine/cryosat_ice2ioda.py:21]: This code shouldn't change.  The CMake build and install processes will be substituting the `@SCRIPT_LIB_PATH@` variable.

**Review by markjolah — CHANGES_REQUESTED** [review]

**Comment by srherbener** [comment]
@mjkagnes123 here are the messages from the CI testing. The cryosat test is passing now, and you are down to just one coding norms error. Thanks for fixing these issues!
```
test 5
--
955 | Start 5: iodaconv_marine_coding_norms
956 |  
957 | 5: Test command: /build_container/bin/iodaconv_lint.sh "/jcsda/ioda-bundle/iodaconv/src/marine" "/jcsda/ioda-bundle/iodaconv"
958 | 5: Environment variables:
959 | 5:  OMP_NUM_THREADS=1
960 | 5: Test timeout computed to be: 10000000
961 | 5: + SRC_DIR=/jcsda/ioda-bundle/iodaconv/src/marine
962 | 5: + PYCODESTYLE_CFG_PATH=/jcsda/ioda-bundle/iodaconv
963 | 5: + cd /jcsda/ioda-bundle/iodaconv/src/marine
964 | 5: + pycodestyle -v --config=/jcsda/ioda-bundle/iodaconv/.pycodestyle --filename=*.py,*.py.in .
965 | 5: cli configuration: /jcsda/ioda-bundle/iodaconv/.pycodestyle
966 | 5: directory .
967 | 5: checking ./argoClim2ioda.py
968 | 5: checking ./cryosat_ice2ioda.py
969 | 5: ./cryosat_ice2ioda.py:62:13: E122 continuation line missing indentation or outdented
970 | 5: checking ./emc_ice2ioda.py
971 | 5: checking ./gds2_sst2ioda.py
972 | 5: checking ./glider2ioda.py
973 | 5: checking ./gmao_obs2ioda.py
974 | 5: checking ./godae_profile2ioda.py
975 | 5: checking ./godae_ship2ioda.py
976 | 5: checking ./godae_trak2ioda.py
977 | 5: checking ./hgodas_adt2ioda.py
978 | 5: checking ./hgodas_insitu2ioda.py
979 | 5: checking ./hgodas_sst2ioda.py
980 | 5: checking ./rads_adt2ioda.py
981 | 5: checking ./smap_sss2ioda.py
982 | 5: checking ./viirs_modis_oc2ioda.py
983 | 5: 1       E122 continuation line missing indentation or outdented
984 | 1/1 Test #5: iodaconv_marine_coding_norms .....***Failed    0.31 sec
985 |  
986 | 0% tests passed, 1 tests failed out of 1
987 |  
988 | Label Time Summary:
989 | iodaconv    =   0.31 sec*proc (1 test)
990 | script      =   0.31 sec*proc (1 test)
991 |  
992 | Total Test time (real) =   0.32 sec
993 |  
994 | The following tests FAILED:
995 | 5 - iodaconv_marine_coding_norms (Failed)
996 | Errors while running CTest
```

**Comment by guillaumevernieres** [comment]
@mjkagnes123 , could you create an issue associated with that PR? 
Thanks @srherbener for reviewing the marine stuff, much appreciated.

File: src/marine/cryosat_ice2ioda.py (hunk 1 above)
Comment (mjkagnes123) [reply src/marine/cryosat_ice2ioda.py]: Thank you for letting me know about it. I brought SCRIPT_LIB_PATH back.

**Comment by mjkagnes123** [comment]
> @mjkagnes123 , could you create an issue associated with that PR?
> Thanks @srherbener for reviewing the marine stuff, much appreciated.

I added issue #457 in the PR. Thank you.

**Comment by mjkagnes123** [comment]
> @mjkagnes123 here are the messages from the CI testing. The cryosat test is passing now, and you are down to just one coding norms error. Thanks for fixing these issues!
I tried to fix the indentation. If there is an issue, would you kindly let me know what exactly CI testing complains about and how I can fix it? Thank you.

**Comment by guillaumevernieres** [comment]
> > @mjkagnes123 here are the messages from the CI testing. The cryosat test is passing now, and you are down to just one coding norms error. Thanks for fixing these issues!
> > I tried to fix the indentation. If there is an issue, would you kindly let me know what exactly CI testing complains about and how I can fix it? Thank you.

@mjkagnes123 , please run the ctests on your side before pushing to github.

**Comment by mjkagnes123** [comment]
I did run ctest before pushing it and ctests for cryosat and marin norm passed.  Still Github shows AWS tests fail.
________________________________
From: guillaumevernieres ***@***.***>
Sent: Thursday, March 25, 2021 7:59 AM
To: JCSDA-internal/ioda-converters ***@***.***>
Cc: Kim, Min-jeong (GSFC-610.1)[MORGAN STATE UNIV.] ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [JCSDA-internal/ioda-converters] Fix to convert multiple CryoSat2 sea-ice freeboard data files into one concatenated daily file (#458)


@mjkagnes123<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmjkagnes123&data=04%7C01%7Cmin-jeong.kim%40nasa.gov%7Cac142a03d3604c1a534d08d8ef857ff7%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637522703952299998%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=CDWvsu2v3OcLRVGvt7aioEyxBoVKGQp4xWfBru7Kg%2BY%3D&reserved=0> here are the messages from the CI testing. The cryosat test is passing now, and you are down to just one coding norms error. Thanks for fixing these issues!
I tried to fix the indentation. If there is an issue, would you kindly let me know what exactly CI testing complains about and how I can fix it? Thank you.

@mjkagnes123<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmjkagnes123&data=04%7C01%7Cmin-jeong.kim%40nasa.gov%7Cac142a03d3604c1a534d08d8ef857ff7%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637522703952299998%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=CDWvsu2v3OcLRVGvt7aioEyxBoVKGQp4xWfBru7Kg%2BY%3D&reserved=0> , please run the ctests on your side before pushing to github.

**Comment by guillaumevernieres** [comment]
Oops, that is annoying @mjkagnes123 . @srherbener , any idea why that would be?

File: src/marine/cryosat_ice2ioda.py (hunk 1 above)
Comment (srherbener) [reply src/marine/cryosat_ice2ioda.py]: @mjkagnes123 thank you for addressing this. @markjolah I'm going to reset your review - please review again. Thanks!

**Comment by srherbener** [comment]
> Oops, that is annoying @mjkagnes123 . @srherbener , any idea why that would be?

I just built and ran this feature branch on my iMac. The coding norms test failed pointing out an indentation problem in the cryosat_ice2ioda.py. I went ahead and fixed this error (it was just one line in the whole file) and after doing that all the tests passed.

I've pushed my fix to this branch, but I'm expecting that the CI tests will fail because they are unfortunately broken now. We are working to get this fixed. My testing on my iMac suggests that this PR is okay now and it's just a matter of getting the CI testing working again.

Thank you for your patience!

**Review by markjolah — APPROVED** [review]
Thanks for fixing the header path name.  From a technical viewpoint this looks OK.  One note is that "cyostat_nc4classic.nc" will not be deleted if an exception is thrown at any point before line 82.  You may wish to address this, by adding `try:` and `finally:` blocks to the for loop.  See the last example on this tutorial page: https://www.programiz.com/python-programming/exception-handling

**Review by srherbener — APPROVED** [review]
Looks good. Thank you for your patience while we have been working through the CI testing issues.

**Pull request merged**